### PR TITLE
Fix some references to 3.2 (and even 3.1) to point to 4.0

### PR DESCRIPTION
**Pull request by paususe**
There are still many more to be fixed, see my mail to galaxy-devel and galaxy-qa. E. g. the migration guide and the retail installation from packages seem to need quite some work.

### CODE REVIEW

File: modules/retail/pages/retail-admin.adoc [hunk 1]
```diff
@@ -2,4 +2,4 @@
 = Administration
 
 This sections contains notes on administering your {smr} installation.
-For general administration tasks, see the {susemgr} documentation at https://www.suse.com/documentation/suse-manager-3/
+For general administration tasks, see the {susemgr} documentation at https://www.suse.com/documentation/suse-manager-4/
```
Comment (Loquacity) [inline modules/retail/pages/retail-admin.adoc]:
```suggestion
For general administration tasks, see the {susemgr} documentation at https://www.suse.com/documentation/suse-manager/
```

File: modules/retail/pages/retail-install-packages.adoc [hunk 2]
```diff
@@ -7,7 +7,7 @@ If you have chosen not to use the Unified Installer, you can install {smr} manua
 
 == Install the {susemgr} server
 
-For instructions to install {susemgr}, see the https://www.suse.com/documentation/suse-manager-3/3.2/susemanager-getting-started/html/book.suma.getting-started/quickstart.chapt.overview.requirements.html#quickstart.sect.introduction
+For instructions to install {susemgr}, see https://www.suse.com/documentation/suse-manager-4/4.0/suse-manager/installation/install-server-unified.html
```
Comment (Loquacity) [inline modules/retail/pages/retail-install-packages.adoc]:
```suggestion
For instructions to install {susemgr}, see xref:installation:install-server-unified.adoc[].
```

File: modules/retail/pages/retail-install-packages.adoc [hunk 3]
```diff
@@ -22,8 +22,9 @@ This functionality will be installed later, after the initial setup.
 == Configure the {susemgr} server
 
 It is important that all required SUSE channels are available on your system and synchronized in order for {smr} to operate correctly.
-For more information on channel management, see https://www.suse.com/documentation/suse-manager-3/3.2/susemanager-advanced-topics/html/book.suma.advanced.topics/suse.mgr.command.line.tools.html#syncing.suse.mgr.repositories.scc
+For more information on channel management, see https://www.suse.com/documentation/suse-manager-4/4.0/suse-manager/client-configuration/channels.html
```
Comment (Loquacity) [inline modules/retail/pages/retail-install-packages.adoc]:
```suggestion
For more information on channel management, see xref:client-configuration:channels.adoc[]
```

File: modules/retail/pages/retail-install-packages.adoc [hunk 4]
```diff
@@ -42,7 +43,7 @@ Channels required for {smr} functionality:
 
 After you have synchronized the required SUSE channels, create channels for any custom software.
 Channels are used to provide custom software for OS image building.
-For more information on software channels, see https://www.suse.com/documentation/suse-manager-3/3.2/susemanager-reference/html/book.suma.reference.manual/ref.webui.channels.html
+For more information on software channels, see https://www.suse.com/documentation/suse-manager-4/4.0/suse-manager/reference/software/software-channel-list-menu.html
```
Comment (Loquacity) [inline modules/retail/pages/retail-install-packages.adoc]:
```suggestion
For more information on software channels, see xref:reference:software/software-channel-list-menu.adoc[]
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 5]
```diff
@@ -19,8 +19,8 @@ Each procedure is detailed in this section.
 
 == Install the build host and register it to {susemgr}
 
-Your build host should be a Salt minion, running {sls}{nbsp}12 SP3.
-For instructions to install Salt minion see https://www.suse.com/documentation/suse-manager-3/3.2/susemanager-getting-started/html/book.suma.getting-started/preparing.and.registering.clients.html
+Your build host should be a Salt minion, running {sls}{nbsp}12 SP4.
+For instructions to install Salt minion see https://www.suse.com/documentation/suse-manager-4/4.0/suse-manager/client-configuration/client-config-overview.html
```
Comment (Loquacity) [inline modules/retail/pages/retail-install-setup.adoc]:
```suggestion
For instructions to install Salt minion see xref:client-configuration:client-config-overview.adoc[].
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 6]
```diff
@@ -50,7 +50,7 @@ Check that the build host has these packages installed after you have run Highst
 - kiwi
 - kiwi-desc-saltboot
 
-If any package is missing, make sure the ``SUSE Manager 3.2 Tools`` repository is available on the build host and install any missing packages manually using ``zypper install``.
+If any package is missing, make sure the ``SUSE Manager 4.0 Tools`` repository is available on the build host and install any missing packages manually using ``zypper install``.
```
Comment (Loquacity) [inline modules/retail/pages/retail-install-setup.adoc]:
```suggestion
If any package is missing, make sure the [package]``SUSE Manager {productnumber} Tools`` repository is available on the build host and install any missing packages manually using [command]``zypper install``.
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 7]
```diff
@@ -68,23 +68,24 @@ Additionally, you will need to create a system group for each branch server, and
 You can create system groups using the {susemgr} {webui}.
 Navigate to menu:Systems[System Groups] and click on btn:[Create System Group].
 
-For more information about system groups, see https://www.suse.com/documentation/suse-manager-3/3.2/susemanager-reference/html/book.suma.reference.manual/ref.webui.systems.systems.html#ref.webui.systems.systemgroups
+For more information about system groups, see https://www.suse.com/documentation/suse-manager-4/4.0/suse-manager/reference/systems/system-groups.html
```
Comment (Loquacity) [inline modules/retail/pages/retail-install-setup.adoc]:
```suggestion
For more information about system groups, see xref:reference:systems/system-groups.adoc[].
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 8]
```diff
@@ -68,23 +68,24 @@ Additionally, you will need to create a system group for each branch server, and
...
 
 
 [[retail.sect.install.branch]]
 == Install the branch server and register it to {susemgr}
 
-Your branch server should be installed as a default Salt-based client ("minion"), running {susemgrproxy} 3.2.
+Your branch server should be installed as a default Salt-based client ("minion"), running {susemgrproxy} 4.0.
```
Comment (Loquacity) [inline modules/retail/pages/retail-install-setup.adoc]:
```suggestion
Your branch server should be installed as a default Salt-based client ("minion"), running {susemgrproxy} {productnumber}.
```

File: modules/retail/pages/retail-install-setup.adoc [hunk 9]
```diff
@@ -68,23 +68,24 @@ Additionally, you will need to create a system group for each branch server, and
...
 
 [WARNING]
 ====
 Do not install the branch server as a traditionally managed client.
 ====
 
-For instructions to install Salt-based proxy minions and register them to {susemgr}, see https://www.suse.com/documentation/suse-manager-3/3.2/susemanager-advanced-topics/html/book.suma.advanced.topics/advanced.topics.proxy.quickstart.html
+For instructions to install Salt-based proxy minions and register them to {susemgr}, see https://www.suse.com/documentation/suse-manager-4/4.0/suse-manager/installation/install-proxy-unified.html
```
Comment (Loquacity) [inline modules/retail/pages/retail-install-setup.adoc]:
```suggestion
For instructions to install Salt-based proxy minions and register them to {susemgr}, see xref:installation:install-proxy-unified.adoc[].
```

File: modules/retail/pages/retail-introduction.adoc [hunk 10]
```diff
@@ -28,9 +28,9 @@ Migrating configuration from {slepos} to {smr} is a technology preview and will
 == About this book
 
 This document provides an overview of {smr}, and guides you through initial installation and setup.
-It should be read in conjunction with the {susemgr} documentation suite, available from https://www.suse.com/documentation/suse-manager-3/
+It should be read in conjunction with the {susemgr} documentation suite, available from https://www.suse.com/documentation/suse-manager-4
```
Comment (Loquacity) [inline modules/retail/pages/retail-introduction.adoc]:
```suggestion
It should be read in conjunction with the {susemgr} documentation suite, available from https://www.suse.com/documentation/suse-manager
```

File: modules/retail/pages/retail-introduction.adoc [hunk 11]
```diff
@@ -28,9 +28,9 @@ Migrating configuration from {slepos} to {smr} is a technology preview and will
...
 
-For more information about managing your {smr} environment, or to find out where to get help, see https://www.suse.com/documentation/suse-manager-for-retail-3-2/retail-getting-started/retail.chap.next.html
+For more information about managing your {smr} environment, or to find out where to get help, see https://www.suse.com/documentation/suse-manager-4/4.0/suse-manager/retail/retail-introduction.html
```
Comment (Loquacity) [inline modules/retail/pages/retail-introduction.adoc]:
```suggestion
For more information about managing your {smr} environment, or to find out where to get help, see xref:retail:retail-introduction.adoc[].
```

File: modules/retail/pages/retail-migration-slepostosuma.adoc [hunk 12]
```diff
@@ -15,7 +15,7 @@ Importing and deploying the converted data can also be done in one step, or mult
 In this procedure, you create a single data dump in an XML file, convert it to YAML, and migrate it to the new infrastructure all at once.
 
 . Install a {susemgr} server {productnumber} (with the Retail pattern).
-For more information, see https://www.suse.com/documentation/suse-manager-for-retail-3-2/retail-getting-started/retail.chap.install.html.
+For more information, see https://www.suse.com/documentation/suse-manager-4/4.0/suse-manager/retail/retail-install.html.
```
Comment (Loquacity) [inline modules/retail/pages/retail-migration-slepostosuma.adoc]:
```suggestion
For more information, see xref:retail:retail-install.adoc[].
```

File: modules/retail/pages/retail-migration-slepostosuma.adoc [hunk 13]
```diff
@@ -44,7 +44,7 @@ If there are duplicate MAC addresses, importing the YAML file will fail.
 [NOTE]
 ====
 {slepos} images will not be migrated. You must rebuild the images using the OS image building functionality.
-For more information about building images, see https://www.suse.com/documentation/suse-manager-3/3.2/susemanager-advanced-topics/html/book.suma.advanced.topics/at.images.html#at.images.kiwi.
+For more information about building images, see https://www.suse.com/documentation/suse-manager-4/4.0/suse-manager/administration/image-management.html.
```
Comment (Loquacity) [inline modules/retail/pages/retail-migration-slepostosuma.adoc]:
```suggestion
For more information about building images, see xref:administration:image-management.adoc[].
```

File: modules/retail/pages/retail-migration-slepostosuma.adoc [hunk 14]
```diff
@@ -71,7 +71,7 @@ After all the branches are migrated, shutdown and remove the old SLEPOS Admin Se
 In this procedure, you migrate the SLEPOS infrastructure and the branches one by one, first exporting and then importing.
 
 . Install a {susemgr} server {productnumber} (with the Retail pattern).
-For more information, see https://www.suse.com/documentation/suse-manager-for-retail-3-2/retail-getting-started/retail.chap.install.html.
+For more information, see https://www.suse.com/documentation/suse-manager-4/4.0/suse-manager/retail/retail-install.html.
```
Comment (Loquacity) [inline modules/retail/pages/retail-migration-slepostosuma.adoc]:
```suggestion
For more information, see xref:retail:retail-install.adoc[].
```

File: modules/retail/pages/retail-migration-slepostosuma.adoc [hunk 15]
```diff
@@ -102,7 +102,7 @@ As long as there are duplicate MAC addresses, {susemgr} will refuse importing th
 [NOTE]
 ====
 {slepos} images will not be migrated. You must rebuild the images using the OS image building functionality.
-For more information about building images, see https://www.suse.com/documentation/suse-manager-3/3.2/susemanager-advanced-topics/html/book.suma.advanced.topics/at.images.html#at.images.kiwi.
+For more information about building images, see https://www.suse.com/documentation/suse-manager-4/4.0/suse-manager/administration/image-management.html.
```
Comment (Loquacity) [inline modules/retail/pages/retail-migration-slepostosuma.adoc]:
```suggestion
For more information about building images, see xref:administration:image-management.adoc[].
```

File: modules/retail/pages/retail-next.adoc [hunk 16]
```diff
@@ -17,7 +17,7 @@ If you need further information or support, try one of these options.
 [[retail.sect.next.docs]]
 === More documentation
 
-For {susemgr} documentation, visit https://www.suse.com/documentation/suse-manager-3/
+For {susemgr} documentation, visit https://www.suse.com/documentation/suse-manager-4/
```
Comment (Loquacity) [inline modules/retail/pages/retail-next.adoc]:
```suggestion
For {susemgr} documentation, visit https://www.suse.com/documentation/suse-manager/
```

File: modules/retail/pages/retail-requirements.adoc [hunk 17]
```diff
@@ -14,7 +14,7 @@ endif::[]
 Before you install {smr}, ensure your environment meets the minimum requirements.
 This section lists the requirements for your {smr} installation.
 These requirements are in addition to the {susemgr} requirements listed at
-https://www.suse.com/documentation/suse-manager-3/3.2/susemanager-getting-started/html/book.suma.getting-started/quickstart.chapt.overview.requirements.html#quickstart.sect.prereq
+https://www.suse.com/documentation/suse-manager-4/4.0/suse-manager/installation/general-requirements.html
```
Comment (Loquacity) [inline modules/retail/pages/retail-requirements.adoc]:
```suggestion
xref:installation:general-requirements.adoc[].
```

Needs to be on the same line as previous, as well.

File: modules/upgrade/pages/sp-migration.adoc [hunk 18]
```diff
@@ -4,9 +4,10 @@
 
 
 
-SUSE Manager uses {sls} 12 as its underlying operating system.
-Therefore Service Pack migration (for example, from version 12 SP1 to 12 SP3) may be performed in the same way as a typical {slsa} migration.
+SUSE Manager uses {sls} 15 SP1 as its underlying operating system.
```
Comment (Loquacity) [inline modules/upgrade/pages/sp-migration.adoc]:
```suggestion
SUSE Manager uses {sls} {sles-version} {sp-version} as its underlying operating system.
```

**Comment by Loquacity** [comment]
Some of these might conflict with https://github.com/SUSE/doc-susemanager/pull/547

As I said on that PR: 

> The full links in the Retail book were there because we didn't have a good way to link between books when this one was originally published. We do now, so we should use it. Also, dropping the version number on suse.com links will redirect to the latest version, so we should omit the version number where we can.

**Comment by paususe** [comment]
> Some of these might conflict with #547
> 
> As I said on that PR:
> 
> > The full links in the Retail book were there because we didn't have a good way to link between books when this one was originally published. We do now, so we should use it. Also, dropping the version number on suse.com links will redirect to the latest version, so we should omit the version number where we can.

I am not sure we should omit version number.

If we omit the version number, the 4.0 documentation will always link to the latest documentation. As 4.0 = latest documentation, it is right now. 

But when SUSE Manager 4.1 is out, the 4.0 documentation will still link to the latest documentation (which will be 4.1), thus creating wrong links and references (from 4.0 to 4.1).

What do you think?

**Comment by Loquacity** [comment]
> > Some of these might conflict with #547
> > As I said on that PR:
> > > The full links in the Retail book were there because we didn't have a good way to link between books when this one was originally published. We do now, so we should use it. Also, dropping the version number on suse.com links will redirect to the latest version, so we should omit the version number where we can.
> 
> I am not sure we should omit version number.
> 
> If we omit the version number, the 4.0 documentation will always link to the latest documentation. As 4.0 = latest documentation, it is right now.
> 
> But when SUSE Manager 4.1 is out, the 4.0 documentation will still link to the latest documentation (which will be 4.1), thus creating wrong links and references (from 4.0 to 4.1).
> 
> What do you think?

If it's important to link to the specific version of a doc, then we should do that intentionally. However, now that we have xrefs, we really should only be linking to a documentation URL if we are talking about the suite as a whole, in which case the most recent will be what's required. There's really no reason to be deep linking to our own documentation at all, now.

**Comment by Loquacity** [comment]
@paususe Is it OK if we close this now please?

**Comment by paususe** [comment]
If the changes were if included in some other PR, sure, go ahead and close this one

**Pull request closed without merging**